packages/javascript-sdk/src/token-manager/index.ts

```diff
@@ -118,9 +118,16 @@ abstract class TokenManager {
     const verifier = PKCE.createVerifier();
     const state = PKCE.createState();
 
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    const { forceRenew, login, ...config } = options;
+    // so to not change the type of the above function
+    // we assign it here if its undefined or null.
+
+    if (!options) options = {};
+
+    // in the scenario above where an options object is undefined or null
+    // and we try to destructure a value from an empty object
+    // we assign defaults
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { forceRenew = false, login = true, ...config } = options;
```

Comment:
I'm okay with this, but if we have an alternative that doesn't require the disabling of rules and other stuff, I think I'd prefer that instead. Is this any better:

```js
const config = Object.assign({}, options);
delete config.forceRenew
delete config.login
```

The above doesn't need the condition, the comments or the rule disabling. I think it's quite self-explanatory. Another plus is we aren't actually using the `forceRenew` and `login` that we are pulling out (hence the rule disabling), we're just using them to get to the rest (no pun intended).